pimd/pim_iface.c

```diff
@@ -1762,3 +1762,63 @@ void pim_iface_init(void)
 	if_zapi_callbacks(pim_ifp_create, pim_ifp_up, pim_ifp_down,
 			  pim_ifp_destroy);
 }
+
+static void pim_if_membership_clear(struct interface *ifp)
+{
+	struct pim_interface *pim_ifp;
+
+	pim_ifp = ifp->info;
+	assert(pim_ifp);
+
+	if (pim_ifp->pim_enable && pim_ifp->gm_enable)
+		return;
+
+	pim_ifchannel_membership_clear(ifp);
+}
+
+int pim_pim_interface_delete(struct interface *ifp)
+{
+	struct pim_interface *pim_ifp = ifp->info;
+
+	if (!pim_ifp)
+		return 1;
```

Comment:
It's always 1, so maybe `void` at all? Or is this a mistake and should be 0?

Reply:
Yeah you are right. I have moved the api from one file to another, did not change this part of code. But I will fix it. I will change the api to void.